Code/Framework/AzCore/AzCore/std/math.h

```diff
@@ -30,4 +30,51 @@ namespace AZStd
     using std::sqrt;
     using std::tan;
     using std::trunc;
+
+} // namespace AZStd
+
+// from c++20 standard
+namespace AZStd::Internal
+{
+    template<typename T>
+    constexpr T lerp(T a, T b, T t) noexcept
+    {
+        if ((a <= 0 && b >= 0) || (a >= 0 && b <= 0))
+        {
+            return t * b + (1 - t) * a;
+        }
+
+        if (t == 1)
+        {
+            return b;
+        }
+        const T x = a + t * (b - a);
+        if ((t > 1) == (b > a))
+        {
+            return b < x ? x : b;
+        }
+        else
+        {
+            return x < b ? x : b;
+        }

```

Comment:
I agree that there is a trade-off here between accuracy and speed.
This should likely involve some higher-level architectural decisions:

* should such a choice be provided for other functions `sin<Fast>` or the famous `fisqrt<Fast>`
* if so, how do we provide the std functions in such a context ? additional namespace instead of template parameter ?

Reply:
It's not unprecedented to have extensions and additions to the standard in AZStd, so it could be considered. Some care has to be taken because for the average game the majority can get away with lower accuracy, but O3DE is also used for other purposes such as simulations and digital twins that would require a higher accuracy. It's a bigger question about what to do with the math functions in the `std::` scope, so maybe an RFC to collect thoughts around this?

Reply:
Could we not have the `AZStd` version just be this implementation, and then the `AZ` version be the fast/simple one? Then we wouldn't need to augment the `std::` version with additional template parameters or anything (we could add a comment too to say, if for some reason this showing up in a profile, consider switching to `AZ::Lerp`?).